rebac-admin-backend/v1/identities_test.go

```diff
@@ -0,0 +1,585 @@
+	)
+
+	mockIdentityGroups := resources.Groups{
+		Data: []resources.Group{
+			{
+				Id:   &mockId,
+				Name: mockName,
+			},
+		},
+	}
+	expectedResponse := resources.GetIdentityGroupsResponse{
+		Data:   mockIdentityGroups.Data,
+		Status: http.StatusOK,
+	}
+
+	params := resources.GetIdentitiesItemGroupsParams{}
+	mockIdentityService := interfaces.NewMockIdentitiesService(ctrl)
+	mockIdentityService.EXPECT().GetIdentityGroups(gomock.Eq(mockIdentityId), gomock.Eq(&params)).Return(&mockIdentityGroups, nil)
+
+	mockWriter := httptest.NewRecorder()
+	mockRequest := httptest.NewRequest(http.MethodGet, fmt.Sprintf("/identities/%s/groups", mockIdentityId), nil)
+
+	sut := handler{Identities: mockIdentityService}
+	sut.GetIdentitiesItemGroups(mockWriter, mockRequest, mockIdentityId, params)
+
+	result := mockWriter.Result()
+	defer result.Body.Close()
+
+	data, err := io.ReadAll(result.Body)
+	if err != nil {
+		t.Errorf("Expected error to be nil, got %v", err)
+	}
+
+	response := new(resources.GetIdentityGroupsResponse)
+	if err := json.Unmarshal(data, response); err != nil {
+		t.Errorf("Unexpected err while unmarshaling GetIdentitiesItemGroups response, got: %v", err)
+	}
+
+	c.Assert(result.StatusCode, qt.Equals, http.StatusOK)
+	c.Assert(response, qt.DeepEquals, &expectedResponse)
+}
+
+func TestHandler_PatchIdentitiesItemGroupsSuccess(t *testing.T) {
+
+}
+
+func TestHandler_GetIdentitiesItemRolesSuccess(t *testing.T) {
+	c := qt.New(t)
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	var (
+		mockId   = "mock-id"
+		mockName = "test-rolename"
+	)
+
+	mockIdentityRoles := resources.Roles{
+		Data: []resources.Role{
+			{
+				Id:   &mockId,
+				Name: mockName,
+			},
+		},
+	}
+	expectedResponse := resources.GetIdentityRolesResponse{
+		Data:   mockIdentityRoles.Data,
+		Status: http.StatusOK,
+	}
+
+	params := resources.GetIdentitiesItemRolesParams{}
+	mockIdentityService := interfaces.NewMockIdentitiesService(ctrl)
+	mockIdentityService.EXPECT().GetIdentityRoles(gomock.Eq(mockIdentityId), gomock.Eq(&params)).Return(&mockIdentityRoles, nil)
+
+	mockWriter := httptest.NewRecorder()
+	mockRequest := httptest.NewRequest(http.MethodGet, fmt.Sprintf("/identities/%s/roles", mockIdentityId), nil)
+
+	sut := handler{Identities: mockIdentityService}
+	sut.GetIdentitiesItemRoles(mockWriter, mockRequest, mockIdentityId, params)
+
+	result := mockWriter.Result()
+	defer result.Body.Close()
+
+	data, err := io.ReadAll(result.Body)
+	if err != nil {
+		t.Errorf("Expected error to be nil, got %v", err)
+	}
+
+	response := new(resources.GetIdentityRolesResponse)
+	if err := json.Unmarshal(data, response); err != nil {
+		t.Errorf("Unexpected err while unmarshaling GetIdentitiesItemRoles response, got: %v", err)
+	}
+
+	c.Assert(result.StatusCode, qt.Equals, http.StatusOK)
+	c.Assert(response, qt.DeepEquals, &expectedResponse)
+}
+
+func TestHandler_PatchIdentitiesItemRolesSuccess(t *testing.T) {
+
+}
+
+func TestHandler_Failures(t *testing.T) {
```

Comment:
I like this kind of table test. We should do the same thing for the happy path tests, and then merge them all in a single test function, because there's no point in having separate test functions.